### PR TITLE
Fixes #613: Add Spyse analyzer | adjusts to env files

### LINKED ISSUE #613: [New analyzer] Spyse.com API
Mark from their marketing team reached out to me on Twitter for this integration a long time back but I forgot about it.

SDK: https://github.com/spyse-com/spyse-python

### PR DESCRIPTION
**Pull request by Palash-Vishnani**
This is the alternate updated PR of #667.

# Checklist

- [x] The pull request is for the branch develop
- [x] A new analyzer or connector was added, in which case:
    - [x] [Usage](https://github.com/intelowlproject/IntelOwl/blob/master/docs/source/Usage.md) file was updated.
    - [ ] (N/A) [Advanced-Usage](./Advanced-Usage.md) was updated (incase the analyzer/connector provides additional optional configuration).
    - [x] Secrets were added in [env_file_app_template](https://github.com/intelowlproject/IntelOwl/blob/master/docker/env_file_app_template), [env_file_app_ci](https://github.com/certego/IntelOwl/blob/master/docker/env_file_app_ci) and in the [Installation](./Installation.md) docs, if necessary.
    - [x] If the analyzer/connector requires mocked testing, `_monkeypatch()` was used in it's class to apply the necessary decorators.
    - [ ] (N/A) If a File analyzer was added, it's name was explicitly defined in [test_file_scripts.py](https://github.com/intelowlproject/IntelOwl/blob/master/tests/analyzers_manager/test_file_scripts.py) (not required for Observable Analyzers).
- [x] (N/A) If external libraries/packages with restrictive licenses were used, they were added in the [ReadMe - Legal Notice](https://github.com/certego/IntelOwl/blob/master/README.md) section.
- [x] The tests gave 0 errors.
- [x] `Black` gave 0 errors.
- [x] `Flake` gave 0 errors.
- [ ] The commits were squashed into a single one (optional, they will be squashed anyway by the maintainer)

### CODE REVIEW

**Comment by Palash-Vishnani** [comment]
@sp35 @eshaan7
Is there anything else that I need to do? Maybe the PR is ready now.

**Review by sp35 — APPROVED** [review]
LGTM, please show me a screenshot or json of results from this analyzer. Thank you.

File: api_app/analyzers_manager/observable_analyzers/spyse.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            if re.match(r"^[\w\.\+\-]+\@[\w]+\.[a-z]{2,3}$", self.observable_name):
+                uri = f"email/{self.observable_name}"
+            else:
+                raise AnalyzerRunException(
+                    f"{self.observable_name} not supported."
+                    "Please enter a valid email address."
+                )
+
+        try:
+            response = requests.get(self.base_url + uri, params=params)
```
[the remark this answers is not in the document]
Comment (eshaan7) [reply api_app/analyzers_manager/observable_analyzers/spyse.py:34]: I'd recommend again raising `AnalyzerRunException` in the `else` case. The thing is the code should _never_ reach this `else` clause in the first place but just so it does we can just raise the exception that `{self.observable_classification} not supported`.

**Comment by Palash-Vishnani** [comment]
@sp35
Actually the thing is my docker isn't working from 2-3 days and I am trying to fix it's problem from soo long. And since my docker isn't running, IntelOwl is also facing problem in my PC. I really apologize for this, I'll try my best to solve the docker's issue, but till is there any other alternative way to test spyse analyzer.....?

**Review by sp35 — CHANGES_REQUESTED** [review]
Please do not ignore the PR instructions next time (re-added the checklist in description). You need to update docs for the new API Key.

File: api_app/analyzers_manager/observable_analyzers/spyse.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+
+class Spyse(classes.ObservableAnalyzer):
+    base_url: str = "https://api.spyse.com/v4/data/"
+
+    def set_params(self, params):
+        self.__api_key = self._secrets["api_key_name"]
+
+    def run(self):
+        params = {"key": self.__api_key}
```
Comment (sp35) [inline api_app/analyzers_manager/observable_analyzers/spyse.py:19]: This is incorrect. The [docs](https://spyse.com/api) say it requires an header instead.

`Authorization: Bearer 776b9620-e645-4f80-81e0-e45eebdade06`

**Review by sp35 — CHANGES_REQUESTED** [review]
@Palash-Vishnani I wanted to test out the changes on your behalf, but there is no free quota available. Maybe @eshaan7 can ask them to share one with you for testing out the integration. 

Till then, please do the changes requested, and let's keep the PR on hold. Thank you!

**Comment by Palash-Vishnani** [comment]
Bravvoooo!!! I am up and running. I'll soon send a commit after updating env_file_app_template, env_file_app_ci and Installation.md docs.
spyse

**Comment by Palash-Vishnani** [comment]
@eshaan7 @sp35 @m0mosenpai 
Guys my docker is working fine. Also my analysis are running but not getting completed, and the output is something like this:

analysis

I am getting this output for every kinds off analysis. Can any of you please suggest me the solution for this and why is this happening??

**Comment by Palash-Vishnani** [comment]
@eshaan7 @sp35 @m0mosenpai 
Guys my docker is working fine. Also my analysis are running but not getting completed, and the output is something like this:

analysis2


I am getting this output for every kinds off analysis though I have the API keys. Can any of you please suggest me the solution for this and why is this happening??

**Comment by eshaan7** [comment]
> @Palash-Vishnani I wanted to test out the changes on your behalf, but there is no free quota available. Maybe @eshaan7 can ask them to share one with you for testing out the integration.
> 
> Till then, please do the changes requested, and let's keep the PR on hold. Thank you!

I have asked them. Let's see what they say.

File: api_app/analyzers_manager/observable_analyzers/spyse.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        if self.observable_classification == self.ObservableTypes.DOMAIN:
+            uri = f"domain/{self.observable_name}"
+        elif self.observable_classification == self.ObservableTypes.IP:
+            uri = f"ip/{self.observable_name}"
+        elif self.observable_classification == self.ObservableTypes.GENERIC:
+            if re.match(r"^[\w\.\+\-]+\@[\w]+\.[a-z]{2,3}$", self.observable_name):
+                uri = f"email/{self.observable_name}"
+            else:
```
Comment (eshaan7) [inline api_app/analyzers_manager/observable_analyzers/spyse.py:27]: under the `generic`, we can also add support for the CVE endpoint. https://spyse-dev.readme.io/reference/cves

**Comment by eshaan7** [comment]
Hey, @Palash-Vishnani. I have just pushed fixes to this PR completing the changes.

**Comment by Palash-Vishnani** [comment]
@eshaan7 
Ohh Ok thanks soo..do I have to do anything else? Modifications or anything?

**Comment by eshaan7** [comment]
> @eshaan7
> Ohh Ok thanks soo..do I have to do anything else? Modifications or anything?

Nope. But feel free to pick up any other issue if you wish to.

**Comment by Palash-Vishnani** [comment]
Hey @eshaan7 the thing is, I thought that my name will get displayed in the contributors list, as my pull request and our commit got merged in this repository by you. But I can't see my name in there, why so?? And sorry to disturb you for this little thing.😅

**Comment by eshaan7** [comment]
> But I can't see my name in there, why so??

The contributors list is picked up from the `master` branch. The latest changes are not in `master` branch yet only `develop`. That's why.

**Pull request merged**